test/plain-list/AutosuggestApp.test.js

```diff
@@ -131,6 +132,102 @@ describe('Plain list Autosuggest', () => {
     });
   });
 
+  describe('when typing and matches exist (when selectFirstSuggestion is true)', () => {
+    before(() => {
+      setSelectFirstSuggestion(true);
+    });
+
+    after(() => {
+      setSelectFirstSuggestion(false);
+    });
+
+    beforeEach(() => {
+      onSuggestionSelected.reset();
+      focusAndSetInputValue('p');
+    });
+
+    it('should focus on any suggestion', () => {
+      expectFocusedSuggestion('Perl');
+    });
+
+    it('should hide suggestions when Escape is pressed', () => {
+      clickEscape();
+      expectSuggestions([]);
+    });
```

Comment:
I think there's a line here and it's your call, so the best strategy here may be for you to merge the PR and remove any tests you think are unnecessary. 

Fwiw though, I went back and forth on this and opted to add all the tests once I noticed that there were actually differences that popped up and needed fixed by having them.

Another option may also be to move the `selectFirstSuggestion = true` tests to a separate file. But as I said, do whatever you think is best.